src/jace/translator/primitive_translators/alu_translator.py

```diff
@@ -0,0 +1,292 @@
+                for (invar, x) in zip(eqn.invars, in_var_names, strict=False)
+                if x is not None
+            )
+
+        else:
+            # This is the general broadcasting case
+            #  We assume that both inputs and the output have the same rank but different sizes in each dimension.
+            #  It seems that Jax ensures this.
+            #  We further assume that if the size in a dimension differs then one must have size 1.
+            #  This is the size we broadcast over, i.e. conceptually replicated.
+            out_shps = tuple(util.get_jax_var_shape(eqn.outvars[0]))  # Shape of the output
+            inp_shpl = tuple(util.get_jax_var_shape(eqn.invars[0]))  # Shape of the left/first input
+            inp_shpr = tuple(
+                util.get_jax_var_shape(eqn.invars[1])
+            )  # Shape of the right/second input
+
+            if not ((len(inp_shpl) == len(inp_shpr)) and (len(out_shps) == len(inp_shpr))):
+                raise NotImplementedError("Can not broadcast over different ranks.")
+
+            for dim, (shp_lft, shp_rgt, out_shp) in enumerate(zip(inp_shpl, inp_shpr, out_shps)):
+                if shp_lft == shp_rgt:
+                    assert out_shp == shp_lft
+                elif shp_lft == 1:
+                    assert shp_rgt == out_shp
+                    dims_to_bcastl.append(dim)
+                elif shp_rgt == 1:
+                    assert shp_lft == out_shp
+                    dims_to_bcastr.append(dim)
+                else:
+                    raise ValueError(f"Invalid shapes in dimension {dim} for broadcasting.")
+
+        # Now we create the Tasklet in which the calculation is performed.
+        tskl_code: str = self._write_tasklet_code(in_var_names, eqn)
+        tskl_name: str = eqn.primitive.name
+        tskl_map_ranges: list[tuple[str, str]] = [
+            (f"__i{dim}", f"0:{N}") for dim, N in enumerate(util.get_jax_var_shape(eqn.outvars[0]))
+        ]
+        tskl_output: tuple[str, dace.Memlet] = None  # type: ignore[assignment]
+        tskl_inputs: list[tuple[str, dace.Memlet] | tuple[None, None]] = []
+
+        # Generate the Memlets for the input.
+        for i, dims_to_bcast in zip(range(len(in_var_names)), [dims_to_bcastl, dims_to_bcastr]):
+            if in_var_names[i] is None:  # Literal: No input needed.
+                tskl_inputs.append((None, None))
+                continue
+            if inp_scalars[i]:  # Scalar
+                assert len(dims_to_bcast) == 0
+                i_memlet = dace.Memlet.simple(in_var_names[i], "0")
+            else:  # Array: We may have to broadcast
+                inputs_: list[str] = []
+                for dim, (map_var, _) in enumerate(tskl_map_ranges):
+                    if dim in dims_to_bcast:
+                        inputs_.append("0")
+                    else:
+                        inputs_.append(map_var)
```

Comment:
This is basically a list comprehension, right? 
```suggestion
                inputs_: list[str] = [
                    "0" if dim in dims_to_bcast else map_var
                    for dim, (map_var, _) in enumerate(tskl_map_ranges)
                ]
```